xarray/core/groupby.py

```diff
@@ -676,6 +682,78 @@ def sizes(self) -> Mapping[Hashable, int]:
             self._sizes = self._obj.isel({self._group_dim: index}).sizes
         return self._sizes
 
+    def shuffle(self, chunks: T_Chunks = None):
+        """
+        Sort or "shuffle" the underlying object.
+
+        "Shuffle" means the object is sorted so that all group members occur sequentially,
+        in the same chunk. Multiple groups may occur in the same chunk.
+        This method is particularly useful for chunked arrays (e.g. dask, cubed).
+        particularly when you need to map a function that requires all members of a group
+        to be present in a single chunk. For chunked array types, the order of appearance
+        is not guaranteed, but will depend on the input chunking.
```

Comment:
IIUC, chunks will then be different sizes from each other. So when writing to Zarr we'll need to re-chunk? (asking for my clarification, feel free to not respond if it's obvious / respond with a single word :) )

Reply:
Yes but that's a zarr limitation :)